lib/ansible/modules/network/junos/junos_user.py

```diff
@@ -202,11 +202,16 @@ def map_obj_to_ele(module, want):
         else:
             operation = 'merge'
 
-        user = SubElement(login, 'user', {'operation': operation})
-
-        SubElement(user, 'name').text = item['name']
+        if item['name'] != 'root':
+            user = SubElement(login, 'user', {'operation': operation})
+            SubElement(user, 'name').text = item['name']
+        else:
+            user = auth = SubElement(element, 'root-authentication', {'operation': operation})
 
         if operation == 'merge':
+            if item['name'] == 'root' and (not item['active'] or item['role'] or item['full_name']):
```

Comment:
@linnil1 Thank you for the PR. Could you please explain what this check is actually doing? I understand, the `active` part. But could you please give a scenario where the check for `item['role'] or item['full_name']` would come into picture?

Reply:
Hi, @NilashishC.
We cannot set `full_name` and `role` to root user, so I check these value for someone who accidentally set it. Do I need to delete these checks so that the configuration is checked by Juniper devices  rather than ansible?

Reply:
Hi @linnil1 Thank you for the explanation. I agree, it's better to keep this check in the module code itself. Could you please modify the failure message to "'root' account cannot be deactivated or be assigned a role and a full name"?

Reply:
Thanks for code reviewing.